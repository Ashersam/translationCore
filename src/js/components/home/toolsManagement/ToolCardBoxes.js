import React from 'react';
import PropTypes from 'prop-types';
import {Checkbox} from 'material-ui';
const checkBoxNames = {
  'kt': 'Key Terms',
  'other': 'Other Terms',
  'names': 'Names'
};

const ToolCardBoxes = ({checks, onChecked, selectedCategories, toolName}) => {
  return (
    <div style={{marginLeft: '6%'}}>
      {
        checks.map((id, index) => (
          <div style={{display: 'flex', alignItems: 'center', marginBottom: 5}} key={index}>
            <Checkbox
              style={{width: 'unset'}}
              iconStyle={{fill: 'black', marginRight: 12}}
              checked={selectedCategories.includes(id)}
              onCheck={(e) => {
                onChecked(id, e.target.checked, toolName);
              }}
            />
            <div>{checkBoxNames[id] || id}</div>
          </div>
        ))
      }
    </div>
  );
};

ToolCardBoxes.propTypes = {
  checks: PropTypes.array.isRequired,
  onChecked: PropTypes.func,
  selectedCategories: PropTypes.array.isRequired,
  toolName: PropTypes.string.isRequired
};

export default ToolCardBoxes;
