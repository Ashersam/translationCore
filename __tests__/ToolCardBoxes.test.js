import React from 'react';
import renderer from 'react-test-renderer';
import ToolCardBoxes from '../src/js/components/home/toolsManagement/ToolCardBoxes';
jest.mock('material-ui/Checkbox');

test.skip('translationWords should have three boxes unchecked', () => {
  const props = {
    toolName: 'translationWords',
    selectedCategories: ['names'],
    checks: ['kt', 'other', 'names'],
    onChecked: jest.fn(() => {})
  };
  const component = renderer.create(
    <ToolCardBoxes {...props}></ToolCardBoxes>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('translationNotes should have many boxes unchecked', () => {
  const props = {
    toolName: 'translationNotes',
    selectedCategories: ['names'],
    availableCategories: 'biggy',
    checks: ['kt', 'other', 'names'],
    onChecked: jest.fn(() => {}),
    bookId: 'mat',
    translate: jest.fn((txt) => {txt})
  };
  const component = renderer.create(
    <ToolCardBoxes {...props}></ToolCardBoxes>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});