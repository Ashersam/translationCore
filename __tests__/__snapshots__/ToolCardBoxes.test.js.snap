// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have three boxes unchecked 1`] = `
<div
  style={
    Object {
      "marginLeft": "6%",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginBottom": 5,
      }
    }
  >
    <div>
      Key Terms
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginBottom": 5,
      }
    }
  >
    <div>
      Other Terms
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginBottom": 5,
      }
    }
  >
    <div>
      Names
    </div>
  </div>
</div>
`;
